src/js/script.js

```diff
@@ -0,0 +1,270 @@
+}
+
+function currentSlide(n) {
+    showSlides(slideIndex = n);
+}
+
+function showSlides(n) {
+    var i;
+    var slides = document.getElementsByClassName("mySlides");
+    var dots = document.getElementsByClassName("dot");
+    if (n > slides.length) {
+        slideIndex = 1;
+    }
+    if (n < 1) {
+        slideIndex = slides.length;
+    }
+    for (i = 0; i < slides.length; i++) {
+        slides[i].style.display = "none";
+    }
+    for (i = 0; i < dots.length; i++) {
+        dots[i].className = dots[i].className.replace(" active", "");
+    }
+    slides[slideIndex - 1].style.display = "block";
+    dots[slideIndex - 1].className += " active";
+}
+
+var carousal = document.getElementById("next");
+setInterval(function() {
+    if (carousal) {
+        carousal.click();
+    }
+}, 5000);
+
+/*================================================================================================ */
+
+/*============================================================================================================== */
+
+// JS for Mobile Category Dropdown
+var ul = null;
+var toggleBtn = '';
+var targ = null;
+var menu = null;
+var list = null;
+window.onload = function() {
+    toggleBtn = document.querySelector('.category-mobile');
+    targ = document.querySelector('.category-block');
+    ul = document.querySelector('.test');
+    menu = document.querySelector('.menu-hamburger');
+    list = document.querySelector('.menu-list');
+    var urlParams = new URLSearchParams(window.location.search);
+    console.log(urlParams.get('category'));
+    toggleBtn.innerHTML = "Categories";
+    //toggleBtn.text = '';
+    if (urlParams.get('category') == null) {
+        toggleBtn.innerHTML = "Categories" + '<span class="caret"></span>';
+    } else {
+        toggleBtn.innerHTML = urlParams.get('category') + '<span class="caret"></span>';
+    }
+
+}
+
+function toggleMenu(e) {
+    list.classList.toggle('disp');
+    e.stopPropagation();
+}
+
+function toggleNav(e) {
+
+    targ.classList.toggle('disp');
+    e.stopPropagation();
+}
+
+function changeText(e, id, name) {
+    console.log(name)
+    toggleBtn.innerHTML = e.target.text;
+    targ.classList.toggle('disp');
+    location.href = "/product/" + id + "?category=" + name;
+    console.log(id);
+    e.stopPropagation();
+
+}
+
+
+//=================================================================================================================
+//=================================================================================================================
+
+//JS for Login Form Validation
+
+function ValidateLogin() {
+    var name = document.getElementById('email').value;
+    var password = document.getElementById('psw').value;
+
+
+    var reg = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
+    var passw = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[^a-zA-Z0-9])(?!.*\s).{6,14}$/;
+    var statusemail = false;
+    var statuspassword = false;
+    if (name == '') {
```

Comment:
I guess we discussed this. Form validations had to be built genrically so that they could be reused